jina/peapods/zmq/__init__.py

```diff
@@ -65,7 +65,7 @@ def __init__(
 
         self.opened_socks.extend([self.in_sock, self.out_sock, self.ctrl_sock])
         if self.in_sock_type == zmq.DEALER:
-            self._send_idle()
+            self._send_idle_to_router()
```

Comment:
I think this send and the option, should be moved outside `send_message`.

Reply:
Ideally yes because if it is a send method it should be static and not bound to an instance.
In this case, it is a bit different since the method has the side effect of counting the requests and the bytes sent. This means, it has access and modifies the members.